lib/s3.persist.stream.js

```diff
@@ -0,0 +1,127 @@
+var AWS = require('aws-sdk');
+var _ = require('underscore');
+
+var stream = require('stream');
+var util = require('util');
+
+var rotateHistoryCount = 3;
```

Comment:
I think this should also be a configuration received from outside - we might want to have different rotation for every use case